pkgs/top-level/all-packages.nix

```
@@ -24622,7 +24622,8 @@ with pkgs;
     inherit (libsForQt5.kdeFrameworks) breeze-icons;
   };
 
-  koreader = callPackage ../applications/misc/koreader {};
+  koreader = callPackage ../applications/misc/koreader/src.nix {};
```

Let's use `default.nix`.
```suggestion
  koreader = callPackage ../applications/misc/koreader {};
```

---

> Do we really have to vendor all of these? I'm sure koreader builds fine with any version of most of these dependencies.

That is my next project!

However, **I think that the default `nix-env -iA koreader` should always upstream-approved versions of koreader's dependencies.**

If we diverge from the versions used by upstream, any quirks or bugs that arise from our version-bumps will create "support blowback" for upstream... people will complain to them about weird bugs that they can't reproduce without realizing that it's really *our* fault for bumping the version of something (say, `glib` or `openssl`) without testing its effect on `koreader`.

My plan is to add a top-level parameter `doNotVendor = [ "pname", "pname", "pname" ...` whose *default value* is `[]` (i.e. vendor everything), and add a third top-level package `koreader-unvendored` which "unvendors" as many things as possible.  But we should not mislead people into thinking that `koreader-unvendored` is koreader.  Also I think it is inevitable that `koreader-unvendored` will become broken quite frequently... I'm not sure I can commit to keeping it unbroken at all times the same way I commit to rapid-response maintainership of `koreader/src.nix` and `vendored.nix`.

> Using a vendored version has the disadvantage of longer build times and untimely security updates.

Yes, and the shorter build times are the main attraction for me, since I'm about to start hacking on koreader... soon my ebook reader will finally have a "totally ignore the touchscreen" mode like all of my dead-tree books do!

---

Go ahead.